src/categories.js

```diff
@@ -259,9 +259,29 @@ function getChildrenRecursive(category, uid, callback) {
 			}
 			Categories.getCategoriesData(children, next);
 		},
-		function (childrenData, next) {
-			childrenData = childrenData.filter(Boolean);
-			category.children = childrenData;
+		function (children, next) {
+			children = children.filter(Boolean);
+			category.children = children;
+
+			async.map(children, function (child, next) {
+				next(null, child.cid);
+			}, next);
+		},
+		function (cids, next) {
+			Categories.hasReadCategories(cids, uid, function (err, data) {
```

Comment:
there is no need to use async.map here.
You can rewrite it as below
```
var cids = children.map(function (child) {
    return child.cid;
});
Categories.hasReadCategories(cids, uid, next);
```

Reply:
done